pysatModelUtils/utils/extract.py

```diff
@@ -24,9 +24,152 @@
+                                        '{:}'.format(iname)]))
+            if iname == obs_alt:
+                # pull out altitude units
+                inst_scale[i] = pyutils.scale_units(obs.meta[iname]['units'],
+                                                    mod.meta[mod_alt]['units'])
+
+                # pull out altitude unit scalar
+                alt_scale = inst_scale[i]
+            else:
+                inst_scale[i] = pyutils.scale_units(obs.meta[iname]['units'],
+                                                    mod_units[i])
+    else:
+        alt_scale = 1.
+
+    # create interpolator
+    interp = interpolate.RegularGridInterpolator(points,
+                                                 np.log(mod[mod_alt].values*alt_scale),
+                                                 bounds_error=False,
+                                                 fill_value=None)
+    # use this interpolator to figure out what altitudes we are at
+    # each loop, use the current estimated path through model expressed in pressure
+    # and interp above to get the equivalent altitude of this path
+    # compare this altitude to the actual instrument altitude
+    # shift the equivalent pressure for the instrument up/down
+    # until difference between altitudes is small
+
+    # log of instrument altitude
+    log_ialt = np.log(obs[obs_alt])
+    # initial difference signal
+    diff = log_ialt*0 + 2.*tol
+    while np.any(np.abs(diff) > tol):
+        # create input array using satellite time/position
+        # replace the altitude coord with the fake tiegcm one
+        coords = []
+        for iscale, coord in zip(inst_scale, obs_coords):
+            if coord == obs_alt:
+                # don't scale altitude-like model coordinate
+                coords.append(obs[mod_coord])
+            else:
+                # scale other dimensions to the model
+                coords.append(obs[coord]/iscale)
+
+        coords.insert(0, obs.index.values.astype(int))
+        # to peform the interpolation we need points
+        # like (x1, y1, z1), (x2, y2, z2)
+        # but we currently have a list like
+        # [(x1, x2, x3), (y1, y2, y3), ...]
+        # convert to the form we need
+        # the * below breaks each main list out, and zip
+        # repeatedly outputs a tuple (xn, yn, zn)
+        sat_pts = [inp for inp in zip(*coords)]
+
+        # altitude pulled out from model
+        orbit_alt = interp(sat_pts)
+        # difference in altitude
+        diff = np.e**orbit_alt - np.e**log_ialt
+        # shift index in inst for model pressure level
+        # in the opposite direction to diff
+        # reduced value by scale, the 'scale height'
+        obs[mod_coord] -= diff/scale
+
+    # achieved model altitude
+    obs[obs_out] = np.e**orbit_alt
+
+
+def instrument_view_through_model(obs, mod, obs_coords, mod_data_names,
+                                  autoscale=False, method='linear'):
```

Comment:
Perhaps method should be a dict that corresponds to obs_coords?

Reply:
RegularGrid is either linear or nearest. It doesn't support a mixture of interpolation types along different coordinate dimensions.

Reply:
Yes, but you could do nearest for some data values and linear for others (as appropriate).  Unless you have a good reason to restrict the user to only using one method on all data types?

Reply:
Added support for different interpolation methods via keyword